frontend/src/core/cells/__tests__/cells.test.ts

```diff
@@ -1276,4 +1278,130 @@ describe("cell reducer", () => {
     const cell = cells[0];
     expect(cell.consoleOutputs).toEqual([STDOUT1, STDOUT2]);
   });
+
+  it("can create a cell using selected code", () => {
```

Comment:
can you put each of these in there own tests and share setup code?
```
describe("can create a cell using selected code"

  beforeEach(() => {
   // setup code

  })

  it("can create when includeSelectionAsInitialCode is true and there is selection", () => {
    ...test one
   })
   ...
});
```

this makes it easier to follow given the same initial setup.

could you also:
1. verify edited state
2. do a test with markdown `includeSelectionAsInitialCode` - i noticed a bug when i add a cell above/below a markdown cell with `includeSelectionAsInitialCode=true`, it says it is edited